engine/runtime/src/main/java/org/enso/interpreter/runtime/data/Vector.java

```diff
@@ -69,19 +69,8 @@ public final Object toArray() {
   @Builtin.WrapException(from = UnsupportedMessageException.class)
   public final Vector slice(long start, long end, InteropLibrary interop)
       throws UnsupportedMessageException {
-    long this_length = length(interop);
-    long slice_start = Math.max(0, start);
-    long slice_end = Math.min(this_length, end);
-
-    if (slice_start >= slice_end) {
-      return new Vector(Array.allocate(0));
-    }
-
-    if ((slice_start == 0) && (slice_end == this_length)) {
-      return this;
-    }
-
-    return new Vector(new ArraySlice(this.storage, slice_start, slice_end));
+    Object slice = ArraySlice.create(storage, start, length(interop), end);
+    return slice == null ? this : new Vector(slice);
```

Comment:
Another place where we need to do this `null` check.

This is begging to be forgotten in some place and lead to NPE leaking into Enso.

Why can't we just inline this piece of logic into the `ArraySlice.create` itself?